1-mandatory/3-slideshow/slideshow.js

```diff
@@ -1 +1,100 @@
 // Write your code here
+const images = [
+  {
+    src: "./assets/cute-cat-a.png",
+  },
+  {
+    src: "./assets/cute-cat-b.jpg",
+  },
+  {
+    src: "./assets/cute-cat-c.jpg",
+  },
+];
+
+//variables
+const backwardBtn = document.querySelector("#backward-btn");
+const forwardBtn = document.querySelector("#forward-btn");
+const image = document.querySelector("#carousel-img");
+const autoBackBtn = document.querySelector("#auto-backward");
+const autoForwardBtn = document.querySelector("#auto-forward");
+const stopBtn = document.querySelector("#stop");
+const showTime = document.querySelector(".show-time");
+
+let ids = 0;
+let interval = 2000;
+let intervalID;
+let intervalTimer;
+
+//eventlisteners
+
+// Get image on load
+document.addEventListener("DOMContentLoaded", getImageOnLoad);
+forwardBtn.addEventListener("click", getNextPic);
+backwardBtn.addEventListener("click", getPreviousPic);
+autoBackBtn.addEventListener("click", showAutoBack);
+autoForwardBtn.addEventListener("click", showAutoForward);
+stopBtn.addEventListener("click", stopAutopInterval);
+
+//functions
+function getImageOnLoad() {
+  image.src = images[0].src;
+}
+
+// get next pic
+function getNextPic() {
+  ids++;
+  if (ids > 2) {
+    ids = 0;
+  }
+  image.src = images[ids].src;
+}
+
+// get previous pic
+function getPreviousPic() {
+  ids--;
+  if (ids < 0) {
+    ids = 2;
+  }
+  image.src = images[ids].src;
+}
+
+// show next images automaticlly
+function showAutoForward() {
+  autoForwardBtn.disabled = true;
+  autoBackBtn.disabled = true;
+  showTimer();
+  intervalID = setInterval(() => {
+    getNextPic();
+  }, interval);
+}
```

Comment:
Really clever way of finding the next and previous page!! 
Something that may be a bit more advance is using the modulo operator documented [here](https://stackoverflow.com/questions/25566960/javascript-go-to-previous-next-page) for going to the previous and next page!